packages/core/src/config.js

```diff
@@ -61,10 +63,38 @@ function buildConfig(configFileOpts, cliOpts) {
     docOptions: getDocOptions(cliOpts, config.docOptions),
     printTypeOptions: gePrintTypeOptions(cliOpts, config.printTypeOptions),
     printer: config.printer,
-    skipDocDirective: getSkipDocDirectives(cliOpts, config),
+    skipDocDirective,
+    customDirective: getCustomDirectives(
+      config.customDirective,
+      skipDocDirective,
+    ),
   };
 }
 
+function getCustomDirectives(
+  customDirectiveOptions = {},
+  skipDocDirective = [],
+) {
+  if (Object.keys(customDirectiveOptions).length == 0) {
+    return customDirectiveOptions;
+  }
+
+  for (const [name, option] of Object.entries(customDirectiveOptions)) {
+    if (skipDocDirective.includes(name)) {
+      delete customDirectiveOptions[name];
+    } else if (
+      hasProperty(option, "descriptor") === false ||
+      typeof option.descriptor !== "function"
+    ) {
+      throw new Error(
+        `Wrong format for plugin custom directive "${name}", it should be {descriptor: (directiveType, constDirectiveType) => String}`,
+      );
+    }
+  }
+
+  return customDirectiveOptions;
```

Comment:
As I recommend the use of `undefined`, then we should stay consistent in the return value
```suggestion
  return Object.keys(customDirectiveOptions).length === 0 ? undefined : customDirectiveOptions;
```